src/app/home/panel/cards/dashboard/budgets-spending/budgets-spending.component.ts

```diff
@@ -0,0 +1,44 @@
+import {Component, OnInit} from '@angular/core';
+import {ApiService} from '../../../../../shared/services/api.service';
+import {saveDocumentWithId} from '../../../../../shared/utilities';
+import {BudgetDto} from '../../../../shared/forms/budgets-form/dtos';
+import {TransactionDto} from '../../../../shared/forms/transaction-form/dtos';
+import {CurrentMonthBudgetSpending} from './dtos';
+
+@Component({
+  selector: 'budgets-spending',
+  templateUrl: './budgets-spending.component.html',
+  styleUrls: ['./budgets-spending.component.scss']
+})
+export class BudgetsSpendingComponent implements OnInit {
+
+  currentMonthBudgetsSpendingList: CurrentMonthBudgetSpending[];
+
+  constructor(private apiService: ApiService) {
+  }
+
+  ngOnInit() {
+    this.createBudgetsSpendingList();
+  }
+
+  createBudgetsSpendingList() {
+    this.apiService.getBudgetsList().subscribe(res => {
+      const budgetsList: BudgetDto[] = saveDocumentWithId(res);
+
+      this.apiService.getTransactionsList().subscribe(res => {
+        const transactionsList: TransactionDto[] = saveDocumentWithId(res);
+
+        const currentMonthBudgets: BudgetDto[] = transactionsList.getCurrentMonthTransactions().sumDuplicatedBudgetsAmounts();
+        this.currentMonthBudgetsSpendingList = Object.keys(currentMonthBudgets)
+          .map(budgetId => (
+            {
+              budgetId: budgetId,
```

Comment:
If key name is the same as value variable name, you don't have to reapeat it. Just go with: `budgetId,`